text/1004-transactions-api.rst

```diff
@@ -651,16 +945,129 @@ There are few problems of this approach:
    network failures would be an issue
 
 
-Keep `transaction` as but Add a Helper for Retrying
----------------------------------------------------
+Keep `transaction` as is but Add a Helper for Retrying
+------------------------------------------------------
 
-The problem with this approach is that it hard to teach using `retry`
+The problem with this approach is that it hard to teach using ``retry``
 when raw transactions "work on my laptop". However, this is somewhat
 alleviated by failure injection.
 
 
+Retry All Single Queries
+------------------------
+
+This specification describes that read-only non-transactional queries
+should be retried automatically.
+
+While it's tempting to retry all ``pool.query`` and ``pool.execute``
+calls (even modifications), it **gives the false sense of security**: no
+"concurrent update" issues are seen.  But it's better to see such error
+and turn the whole block of code into a transaction rather than just a
+mutation. I.e. retrying a single mutable request on a "concurrent
+update" error must be a deliberate decision.
```

Comment:
This whole paragraph is very hard to parse. I suggest to reword it to:

"Automatically retrying `query()`-like calls that execute queries with side effects is not always safe. Therefore our recommendation to users would be to use the new `retry()` API when they know that the query would be safe to repeat."

Reply:
This is not enough in my opinion. Added an example for clarification.